app/events/email-incoming/index.ts

```diff
@@ -51,6 +53,42 @@ const cognito = new CognitoIdentityProviderClient({})
 const s3 = new S3Client({})
 const origin = getOrigin()
 
+const legacyAddress = 'mailnull@capella2.gsfc.nasa.gov'
+const legacyFromNameSuffix = ' via gcncirc'
+
+/**
+ * Parse rewritten From addresses from capella2.
+ *
+ * Messages forwarded by capella2 from non-NASA addresses have From headers
+ * that are rewritten like this:
+ *
+ * From: Albert Einstein <albert.einstein@ligo.org>
+ * --rewritten to--
+ * From: "Albert Einstein <albert.einstein@ligo.org> via gcncirc" <mailnull@capella2.gsfc.nasa.gov>
+ *
+ */
+export function getFromAddress(fromAddressObject?: AddressObject) {
```

Comment:
Oh right! I exported this so that I could write a unit test for it. I forgot to write that test. Will do shortly.